src/globus_sdk/exc/api.py

```diff
@@ -0,0 +1,167 @@
+import logging
+from typing import Any, Dict, Optional, cast
+
+from .base import GlobusError
+from .err_info import ErrorInfoContainer
+
+log = logging.getLogger(__name__)
+
+
+class GlobusAPIError(GlobusError):
+    """
+    Wraps errors returned by a REST API.
+
+    :ivar http_status: HTTP status code (int)
+    :ivar code: Error code from the API (str),
+                or "Error" for unclassified errors
+    :ivar message: Error message from the API. In general, this will be more
+                   useful to developers, but there may be cases where it's
+                   suitable for display to end users.
+    """
+
+    MESSAGE_FIELDS = ["message", "detail"]
+    RECOGNIZED_AUTHZ_SCHEMES = ["bearer", "basic", "globus-goauthtoken"]
+
+    def __init__(self, r, *args, **kw):
+        self.http_status = r.status_code
+        # defaults, may be rewritten during parsing
+        self.code = "Error"
+        self.message = r.text
+
+        self._info: Optional[ErrorInfoContainer] = None
+        self._underlying_response = r
+        self._parse_response()
+        super().__init__(*self._get_args())
+
+    @property
+    def raw_json(self):
+        """
+        Get the verbatim error message received from a Globus API, interpreted
+        as JSON data
+
+        If the body cannot be loaded as JSON, this is None
+        """
+        r = self._underlying_response
+        if "Content-Type" in r.headers and (
+            "application/json" in r.headers["Content-Type"]
+        ):
```

Comment:
This could also "if NOT thing, return none; now main block here, de-dented." (I get that it's a bit of a subjective thing, especially in smaller chunks like here.)

Reply:
Yeah, I tend towards that style as well. I think the flatter we can manage to make things, the better. I'll do this one for sure.

Reply:
Writing it this way leads me to add
```python
    def _json_content_type(self) -> bool:
        r = self._underlying_response
        return "Content-Type" in r.headers and (
            "application/json" in r.headers["Content-Type"]
        )
```

so that it can be `if not self._json_content_type(): ...` and read more easily